src/use_case/signup/SignupInteractor.java

```diff
@@ -0,0 +1,36 @@
+package use_case.signup;
+
+import entity.User;
+import interface_adapter.menu.MenuViewModel;
+
+public class SignupInteractor implements SignupInputBoundary {
+    final SignupUserDataAccessInterface userDataAccessObject;
+    final SignupOutputBoundary userPresenter;
+
+    public SignupInteractor(SignupUserDataAccessInterface signupDataAccessInterface,
+                            SignupOutputBoundary signupOutputBoundary) {
+        this.userDataAccessObject = signupDataAccessInterface;
+        this.userPresenter = signupOutputBoundary;
+    }
+
+    @Override
+    public void execute(SignupInputData signupInputData) {
```

Comment:
We should add an extra check to make sure that the password isn't null, I'll open a bug report when this gets merged.